modular_skyrat/modules/death_consequences_perk/death_consequences_trauma.dm

```diff
@@ -0,0 +1,525 @@
+	var/eigenstasium_degradation_decrease = DEATH_CONSEQUENCES_DEFAULT_EIGENSTASIUM_DEGRADATION_REDUCTION
+
+	/// When sansufentanyl is metabolized, degradation is reduced by this.
+	var/sansufentanyl_degradation_decrease = DEATH_CONSEQUENCES_DEFAULT_SANSUFENTANYL_DEGRADATION_REDUCTION
+
+	/// If our victim is dead, their passive degradation will be multiplied against this if they are in stasis.
+	var/on_stasis_death_degradation_mult = 0
+
+	/// If true, when [current_degradation] reaches [max_degradation], we will DNR and ghost our victim.
+	var/permakill_if_at_max_degradation = FALSE
+	/// If true, when [current_degradation] reaches [max_degradation], we will DNR and KILL our victim.
+	var/force_death_if_permakilled = FALSE
+
+	/// If we have killed our owner permanently.
+	var/final_death_delivered = FALSE
+
+	// Higher = overall less intense threshold reduction but it still maxes out once it gets there
+	/// The degradation we will begin reducing the crit threshold at.
+	var/crit_threshold_min_degradation = 0
+	/// The degradation we will stop reducing the crit threshold at.
+	var/crit_threshold_max_degradation = 200
+	/// The amount our victims crit threshold will be reduced by at [crit_threshold_max_degradation] degradation.
+	var/max_crit_threshold_reduction = 100
+
+	/// The degradation we will begin applying stamina damage at.
+	var/stamina_damage_minimum_degradation = 100
+	/// The degradation we will stop increasing the stamina damage at.
+	var/stamina_damage_max_degradation = 500
+	/// The amount our victims crit threshold will be reduced by at [stamina_damage_max_degradation] degradation.
+	var/max_stamina_damage = 80
+
+	/// Used for updating our crit threshold reduction. We store the previous value, then subtract it from crit threshold, to get the value we had before we adjusted.
+	var/crit_threshold_currently_reduced_by = 0
+
+	/// The last world.time we sent a message to our owner reminding them of their current degradation. Used for cooldowns and such.
+	var/time_of_last_message_sent = -DEATH_CONSEQUENCES_TIME_BETWEEN_REMINDERS
+	/// The time between each reminder ([degradation_messages]).
+	var/time_between_reminders = DEATH_CONSEQUENCES_TIME_BETWEEN_REMINDERS
+
+	/// The current level of degradation. Used mostly for reminder messages.
+	var/current_degradation_level = DEGRADATION_LEVEL_NONE
+
+	// Will be iterated through sequentially, so the higher a path is, the quicker it'll be searched
+	// Make sure to put the larger bonuses and more specific types higher than the generic ones
+	/// A assoc list of (atom/movable typepath -> mult), where mult is used as a multiplier against passive living degradation reduction.
+	var/static/list/buckled_to_recovery_mult_table = list(
+		/obj/structure/bed/medical = 5,
+		/obj/structure/bed = 3,
+
+		/obj/structure/chair/comfy = 2,
+		/obj/structure/chair/sofa = 2,
+		/obj/structure/chair = 1.5,
+
+		/mob/living = 1.25, // being carried
+	)
```

Comment:
I'm kind of not a fan of how this is handled, this feels really weird to me.

Reply:
any suggestions for an alternate implementation?